ir/ir.def

```diff
@@ -60,6 +60,7 @@ class ParserState : ISimpleNamespace, Declaration, IAnnotated {
     optional inline IndexedVector<StatOrDecl>   components;
     // selectExpression can be a SelectExpression, or a PathExpression representing a state
     NullOK Expression                   selectExpression;
+    optional std::vector<ParserState> collapsedChain;
```

Comment:
I think this kind of information belongs in an annotation.

Reply:
It's not easy to put the original IR into an Annotation. One benefit of this approach is that it allows you to easily uncollapse the chain back into its original states or have better visibility to what the IR originally looked like.

Reply:
The state will carry all the original states in its representation, all the visitors will continue to analyze and transform these states, although they no longer technically exist in the program. I expect this will cause some serious maintenance problems in the future. Who needs better visibility into the original IR? Is this for debugging by humans? Or you want to write some visitor that undoes the tranformation later? We have backtracking for this kind of stuff.

Reply:
If the IR was stored in an `IR::Vector`, then yes, those states would be visited and cause the issue you are describing. However, IR in a `std::vector` will not get visited by visitors, so I don't think that problem applies here.

I do want to write a pass that can undo this transformation, yes.

Reply:
I observed that the states stored in my proposed changes no longer appear in the P4 program after the SimplifyParsers pass.